httpevents/handler.go

```diff
@@ -0,0 +1,79 @@
+package httpevents
+
+import (
+	"net/http"
+
+	"github.com/segmentio/events"
+)
+
+// NewHandler wraps the HTTP handler with and access logger which generates
+// events for all incoming requests using logger.
+func NewHandler(logger *events.Logger, handler http.Handler) http.Handler {
+	if logger == nil {
+		logger = events.DefaultLogger
+	}
+	return http.HandlerFunc(func(res http.ResponseWriter, req *http.Request) {
+		w := &responseWriter{
+			ResponseWriter: res,
+			// We capture all the values we need from req in case the object
+			// gets modified by the handler.
+			logger:   logger,
+			address:  req.RemoteAddr,
+			method:   req.Method,
+			host:     req.Host,
+			path:     req.URL.Path,
+			query:    req.URL.RawQuery,
+			fragment: req.URL.Fragment,
+			agent:    req.UserAgent(),
+		}
+
+		// If the handler panics we want to make sure we report the issue in the
+		// access log, while also ensuring that a response is going to be sent
+		// down to the client.
+		// We don't silence the panic here tho and instead we forward it back to
+		// the parent handler which may need to be aware that a panic occurred.
+		defer func() {
+			if err := recover(); err != nil {
+				w.WriteHeader(http.StatusInternalServerError)
```

Comment:
Not sure if we need to be the ones writing this header. Wouldn't rethrowing do this for us?